sympy/polys/multivariate_resultants.py

```diff
@@ -0,0 +1,338 @@
+                               for c in coefficients])
+
+        keep = [column for column in range(dixon_matrix.shape[-1])
+                if any([element != 0 for element in dixon_matrix[:, column]])]
+        return dixon_matrix[:, keep]
+
+class MacaulayResultant():
+    """
+    A class for calculating the Macaulay resultant. Note that the coefficients
+    of the polynomials must be given as symbols.
+
+    Examples
+    ========
+    >>> from sympy.core import symbols
+
+    >>> from sympy.polys.multivariate_resultants import MacaulayResultant
+    >>> x, y, z = symbols('x, y, z')
+
+    >>> a_0, a_1, a_2 = symbols('a_0, a_1, a_2')
+    >>> b_0, b_1, b_2 = symbols('b_0, b_1, b_2')
+    >>> c_0, c_1, c_2,c_3, c_4 = symbols('c_0, c_1, c_2, c_3, c_4')
+
+    >>> f = a_0 * y -  a_1 * x + a_2 * z
+    >>> g = b_1 * x ** 2 + b_0 * y ** 2 - b_2 * z ** 2
+    >>> h = c_0 * y - c_1 * x ** 3 + c_2 * x ** 2 * z - c_3 * x * z ** 2 + c_4 * z ** 3
+
+    >>> mac = MacaulayResultant(polynomials=[f, g, h], variables=[x, y, z])
+    >>> mac.get_monomials_set()
+    >>> matrix = mac.get_matrix()
+    >>> submatrix = mac.get_submatrix(matrix)
+    >>> submatrix
+    Matrix([
+    [-a_1,  a_0,  a_2,    0],
+    [   0, -a_1,    0,    0],
+    [   0,    0, -a_1,    0],
+    [   0,    0,    0, -a_1]])
+
+    Reference
+    ==========
+    1. [Bruce97]_
+    2. [Stiller96]_
+
+    See Also
+    ========
+    Notebook in examples: sympy/example/notebooks.
+    """
+    def __init__(self, polynomials, variables):
+        """
+        Parameters
+        ----------
+        variables: list
+            A list of all n variables
+        polynomials : list of sympy polynomials
+            A  list of m n-degree polynomials
+        """
+        self.polynomials = polynomials
+        self.variables = variables
+        self.n = len(variables)
+
+        # A list of the d_max of each variable.
+        self.degrees = [total_degree(poly, *self.variables) for poly in self.polynomials]
+
+        self.degree_m = self.get_degree_m()
+        self.monomials_size = self.get_size()
+
+    def get_degree_m(self):
+        r"""
+        Returns
+        -------
+        degree_m: int
+            The degree_m is calculated as  1 + \sum_1 ^ n (d_i - 1), where
+            d_i is the degree of the i polynomial
+        """
+        return 1 + sum(d - 1 for d in self.degrees)
+
+    def get_size(self):
+        r"""
+        Returns
+        -------
+        size: int
+            The size of set T. Set T is the set of all possible monomials of
+            the n variables for degree equal to the degree_m
+        """
+        return binomial(self.degree_m + self.n - 1, self.n - 1)
+
+    def get_monomials_of_certain_degree(self, degree):
+        """
+        Returns
+        -------
+        monomials: list
+            A list of monomials of a certain degree.
+        """
+        monomials = list(itermonomials(self.variables, degree) -
```

Comment:
This could be implemented more efficiently by direct use of `combinations_with_replacement` as in `itermonomials` but without adding 1 to the list of variables.